src/vm/types/mod.rs

```diff
@@ -330,31 +554,126 @@ impl Value {
-        Ok(Value::Buffer(BuffData { data: buff_data }))
+        Ok(Value::Sequence(SequenceData::Buffer(BuffData { data: buff_data })))
     }
 
     pub fn buff_from_byte(byte: u8) -> Value {
-        Value::Buffer(BuffData { data: vec![byte] })
+        Value::Sequence(SequenceData::Buffer(BuffData { data: vec![byte] }))
+    }
+
+    pub fn string_ascii_from_bytes(bytes: Vec<u8>) -> Result<Value> {
+        for b in bytes.iter() {
+            if !b.is_ascii_alphanumeric() && !b.is_ascii_punctuation() && !b.is_ascii_whitespace() {
+                return Err(CheckErrors::InvalidCharactersDetected.into());
+            }
+        }
+        // check the string size
+        BufferLength::try_from(bytes.len())?;
+        // construct the string        
+        Ok(Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData { data: bytes }))))
+    }
+
+    pub fn string_utf8_from_string_utf8_literal(tokenized_str: String) -> Result<Value> {
+        let wrapped_codepoints_matcher = Regex::new("^\\\\u\\{(?P<value>[[:xdigit:]]+)\\}").unwrap();
+        let mut window = tokenized_str.as_str();
+        let mut cursor = 0;
+        let mut data: Vec<Vec<u8>> = vec![];
+        while !window.is_empty() {
+            if let Some(captures) = wrapped_codepoints_matcher.captures(window) {
+                let matched = captures.name("value").unwrap();
+                let scalar_value = window[matched.start()..matched.end()].to_string();
+                let unicode_char = {
+                    let u = u32::from_str_radix(&scalar_value, 16).unwrap();
+                    let c = char::from_u32(u).unwrap();
+                    let mut encoded_char: Vec<u8> = vec![0; c.len_utf8()];
+                    c.encode_utf8(&mut encoded_char[..]);
+                    encoded_char
+                };
+
+                data.push(unicode_char);
+                cursor += scalar_value.len() + 4;
+            } else {
+                let ascii_char = window[0..1].to_string().into_bytes();
+                data.push(ascii_char);
+                cursor += 1;
+            }
+            window = &tokenized_str[cursor..];
+        }
+        // check the string size
+        StringUTF8Length::try_from(data.len())?;
+        // construct the string        
+        Ok(Value::Sequence(SequenceData::String(CharType::UTF8(UTF8Data { data }))))
+    }
+
+    pub fn string_utf8_from_bytes(bytes: Vec<u8>) -> Result<Value> {
+        let validated_utf8_str = match str::from_utf8(&bytes) {
+            Ok(string) => string,
+            _ => return Err(CheckErrors::InvalidCharactersDetected.into())
+        };
+        let mut data = vec![];
+        for char in validated_utf8_str.chars() {
+            let mut encoded_char: Vec<u8> = vec![0; char.len_utf8()];
+            char.encode_utf8(&mut encoded_char[..]);
+            data.push(encoded_char);
+        }
+        Ok(Value::Sequence(SequenceData::String(CharType::UTF8(UTF8Data { data }))))
```

Comment:
I think the string size needs to be checked in this constructor.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1779/commits/6677278149912738bfe8f7e74c758a7fc4c24d97